lib/github_changelog_generator/gitlab_fetcher.rb

```diff
@@ -0,0 +1,449 @@
+    CHANGELOG_GITHUB_TOKEN = "CHANGELOG_GITHUB_TOKEN"
+    GH_RATE_LIMIT_EXCEEDED_MSG = "Warning: Can't finish operation: GitHub API rate limit exceeded, changelog may be " \
+    "missing some issues. You can limit the number of issues fetched using the `--max-issues NUM` argument."
+    NO_TOKEN_PROVIDED = "Warning: No token provided (-t option) and variable $CHANGELOG_GITHUB_TOKEN was not found. " \
+    "This script can make only 50 requests to GitHub API per hour without token!"
+
+    # @param options [Hash] Options passed in
+    # @option options [String] :user Gitlab username
+    # @option options [String] :project Gitlab project
+    # @option options [String] :since Only issues updated at or after this time are returned. This is a timestamp in ISO 8601 format: YYYY-MM-DDTHH:MM:SSZ. eg. Time.parse("2016-01-01 10:00:00").iso8601
+    # @option options [Boolean] :http_cache Use ActiveSupport::Cache::FileStore to cache http requests
+    # @option options [Boolean] :cache_file If using http_cache, this is the cache file path
+    # @option options [Boolean] :cache_log If using http_cache, this is the cache log file path
+    def initialize(options = {})
+      @options      = options || {}
+      @user         = @options[:user]
+      @project      = @options[:project]
+      @since        = @options[:since]
+      @http_cache   = @options[:http_cache]
+      @cache_file   = nil
+      @cache_log    = nil
+      @commits      = []
+      @compares     = {}
+
+      Gitlab.sudo = nil
+      @client = Gitlab::Client.new(gitlab_options)
+      @project_id = find_project_id
+    end
+
+    def find_project_id
+      project_id = nil
+      @client.project_search(@project).auto_paginate do |project|
+        if project.namespace.name.eql? @user
+          project_id = project.id
+        end
+      end
+      project_id
+    end
+
+    def gitlab_options
+      result = {}
+      access_token = fetch_github_token
+      result[:private_token] = access_token if access_token
+      endpoint = @options[:github_endpoint]
+      result[:endpoint] = endpoint if endpoint
+      result
+    end
+
+    DEFAULT_REQUEST_OPTIONS = { per_page: PER_PAGE_NUMBER }
+
+    # Fetch all tags from repo
+    #
+    # @return [Array <Hash>] array of tags
+    def get_all_tags
+      print "Fetching tags...\r" if @options[:verbose]
+
+      check_response { fetch_tags }
+    end
+
+    # Fill input array with tags
+    #
+    # @return [Array <Hash>] array of tags in repo
+    def fetch_tags
+      tags = []
+
+       @client.tags(@project_id, DEFAULT_REQUEST_OPTIONS).auto_paginate do |new_tag|
+        tags.push(new_tag)
+      end
+      print_empty_line
+
+      if tags.count == 0
+        Helper.log.warn "Warning: Can't find any tags in repo. \
+Make sure, that you push tags to remote repo via 'git push --tags'"
+      else
+        Helper.log.info "Found #{tags.count} tags"
+      end
+      # tags are a Sawyer::Resource. Convert to hash
```

Comment:
Q: For GitLab content, _are_ the `tag` items really of type `Sawyer::Resource`?

Reply:
Well spotted, they are of type `Gilab::ObjectifiedHash`. Removed the obsolete comment